packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -194,6 +197,9 @@ function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
   );
 
   const sortingOrder: readonly GridSortDirection[] = colDef.sortingOrder ?? rootProps.sortingOrder;
+  const showSortIcon =
+    (colDef.sortable || sortColumnLookup[colDef.field]?.sortDirection != null) &&
```

Comment:
```suggestion
    (colDef.sortable || sortDirection != null) &&
```

There already is sortDirection in the props